src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -723,6 +804,26 @@ void FancyZones::MoveSizeUpdateInternal(HMONITOR monitor, POINT const& ptScreen,
     }
 }
 
+void FancyZones::UpdateVirtualDesktopIDs() noexcept
+{
+    gLock.lock();

```

Comment:
That doesn't solve the data race, since FanceZones is being accessed from multiple threads itself, and we have 2 different mutexes. Please use FanceZones::m_lock instead and remove all global variables.

Reply:
Moving everything out of `FancyZones` class is one step closer to having this as a standalone class somewhere in common, and that's why I did this. I can pass mutex from `FancyZones` to this function and use it as a lock, whould that solve it ?

Reply:
We are moving VirtualDesktopTracker out of FancyZones and into common, but UpdateVirtualDesktopIDs will remain here and it can use FanceZones::m_lock directly without having to pass it.

Reply:
it's still a member function, so no need to pass it explicitly yet. just add `std::unique_lock lock{m_lock};` at the top

Reply:
OK, `UpdateVirtualDesktopIDs` is a member function and can use `m_lock`, but I'm talking about sinchronization between this `UpdateVirtualDesktopIDs` and global function `VirtualDesktopTracker` that is running in separate thread, detecting changes in regkey.

Reply:
Those two functions are both accessing `unordered_set<GUID> gVirtualDesktops`

Reply:
Let's make it a member of `FancyZones` for now and use the same mutex, and do the refactoring in a separate PR. Having a global variable which is being accessed by multiple threads is not ideal.

Reply:
OK, I'll revert to previous solution where this tracking is done inside `FancyZones`.